server/block/purpur.go

```diff
@@ -0,0 +1,64 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+// Purpur is a decorative block that is naturally generated in End cities and End ships.
+type Purpur struct {
+	solid
+	bassDrum
+
+	// Pillar specifies if the block is the pillar variant or not.
+	Pillar bool
+
+	// Axis is the axis which the purpur pillar block faces.
+	Axis cube.Axis
+}
```

Comment:
I would consider splitting this up into `Purpur` and `PurpurPillar`, because a normal purpur block cannot have an Axis. It's a bit counter intuitive to have this available in that case.